### PR TITLE
Rank test script

### PR DESCRIPTION
**Pull request by harrisonpim**
- small part of https://github.com/wellcomecollection/platform/issues/5350

`yarn test` now has a more coherent set of arguments for being run in CI

Rank can also be run from the CLI with a friendly interface for humans (`yarn rank`), which will prompt the user to supply any arguments which they haven't already. `yarn rank` then calls `yarn test` with those collected args

### CODE REVIEW

File: rank/scripts/createIndex.ts [hunk 1]
```diff
@@ -78,20 +72,17 @@ async function go() {
       body: {
         source: {
           index: localIndex,
+          size: 100, // batch size reduced to avoid memory issues
```
Comment (alexwlchan) [inline rank/scripts/createIndex.ts:75]: reduced from what?

File: rank/scripts/createIndex.ts [hunk 2]
```diff
@@ -78,20 +72,17 @@ async function go() {
...
-    if (reindexResp.acknowledged) {
-      success('Reindex started successfully')
-      info('You can monitor the reindex task by running:')
-      code('yarn checkReindex')
-    } else {
-      error('Failed to start reindex with error:')
-      console.info(reindexResp.error)
-    }
+
+    success('Reindex started successfully')
```
Comment (alexwlchan) [inline rank/scripts/createIndex.ts:83]: Why have you removed the error handling code?

File: rank/scripts/rank.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+import { tests as possibleTests } from '../data/tests'
+
+global.fetch = require('node-fetch')
+
+async function go() {
+  const indices: Index[] = await listIndices()
+
+  const possibleTestIds = Object.values(possibleTests)
+    .flatMap((testSet) => testSet.map((test) => test.id))
+    .filter((v, i, a) => a.indexOf(v) == i) // keep unique
```
Comment (alexwlchan) [inline rank/scripts/rank.ts:15]: What are these values? Why does filtering them keep them unique?

**Review by alexwlchan — APPROVED** [review]

File: rank/scripts/createIndex.ts (hunk 1 above)
Comment (harrisonpim) [reply rank/scripts/createIndex.ts]: longer comment added in https://github.com/wellcomecollection/catalogue-api/pull/310/commits/e6abd8e59aaa92d686c2afa1783fe4b6faebbe4d !

File: rank/scripts/rank.ts (hunk 3 above)
Comment (harrisonpim) [reply rank/scripts/rank.ts]: longer comment added in https://github.com/wellcomecollection/catalogue-api/pull/310/commits/e6abd8e59aaa92d686c2afa1783fe4b6faebbe4d !

File: rank/scripts/createIndex.ts (hunk 2 above)
Comment (harrisonpim) [reply rank/scripts/createIndex.ts:83]: The error handling just wasn't working here - starting a reindex successfully would throw errors, etc.
I'm planning to revisit these scripts next week to make wider use of the `gatherArgs()` function and tighten up error handling across the board. Won't be gone for long!

**Pull request merged**